### PR TITLE
Add elapsed_time to Trainer and LogReport

### PR DESCRIPTION
**Pull request by beam2d**
This PR adds `elapsed_time` property to Trainer. This property returns the elapsed time in seconds since the training began. It properly accumulates the elapsed time even if user resumes the training from snapshot.

It also adds `elapsed_time` entry to the log taken by LogReport, and usages in MNIST examples.

Note: an extension to record the elapsed time was originally proposed by @iwiwi in #1472. At that time, I discussed with him and concluded that the elapsed-time recording should be the standard feature of Trainer, which is realized by the current PR.

### CODE REVIEW

**Comment by gwtnb** [comment]
flake8

**Comment by beam2d** [comment]
sorry, fixed

**Comment by okuta** [comment]
Please add test.
But, maybe it is difficult..

File: chainer/training/trainer.py [hunk 1]
```diff
@@ -135,8 +137,25 @@ def __init__(self, updater, stop_trigger=None, out='result'):
         self._done = False
         self._extensions = collections.OrderedDict()
 
+        self._start_at = None
+        self._snapshot_at = 0.0
```
Comment (okuta) [inline chainer/training/trainer.py]: How about `_snapshot_elapsed_time`?

**Comment by beam2d** [comment]
@okuta I fixed the attribute name and added a small test. Would you check them?

**Comment by okuta** [comment]
The test is OK.
Please fix coding style.

**Comment by okuta** [comment]
LGTM except flake8.

**Comment by beam2d** [comment]
Oops, sorry, fixed.

**Comment by okuta** [comment]
Thank you

**Pull request merged**